services-directions-models/src/main/java/com/mapbox/api/directions/v5/utils/ParseUtils.java

```diff
@@ -0,0 +1,134 @@
+package com.mapbox.api.directions.v5.utils;
+
+import com.mapbox.geojson.Point;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Methods to convert Strings to Lists of objects.
+ *
+ */
+public class ParseUtils {
+  private static final String SEMICOLON = ";";
+  private static final String COMMA = ",";
+
+  /**
+   * Parse a String to a list of Integers.
+   * @param original an original String.
+   * @return List of Integers
+   */
+  public static List<Integer> parseToIntegers(String original) {
+    List<Integer> integers = new ArrayList<>();
+    if (original != null) {
+      String[] strings = original.split(SEMICOLON);
+      for (String index : strings) {
+        if (index != null && !index.isEmpty()) {
+          integers.add(Integer.valueOf(index));
+        }
+      }
+    }
+
+    return integers;
+  }
+
+  /**
+   * Parse a String to a list of Strings.
+   * @param original an original String.
+   * @return  List of Strings
+   */
+  public static List<String> parseToStrings(String original) {
+    List<String> strings = new ArrayList<>();
+    if (original != null) {
+      String[] values = original.split(SEMICOLON);
+      for (String name : values) {
+        if (name != null && !name.isEmpty()) {
+          strings.add(name);
+        }
+      }
+    }
+
+    return strings;
+  }
+
+  /**
+   * Parse a String to a list of Points.
+   * @param original an original String.
+   * @return List of Points
+   */
+  public static List<Point> parseToPoints(String original) {
+    List<Point> points = new ArrayList<>();
+    if (original != null) {
+      String[] targets = original.split(SEMICOLON);
+      for (String target : targets) {
+        if (target != null && !target.isEmpty()) {
+          String[] point = target.split(COMMA);
+          points.add(Point.fromLngLat(Double.valueOf(point[0]), Double.valueOf(point[1])));
+        }
+      }
+    }
+
+    return points;
+  }
+
+  /**
+   * Parse a String to a list of Points.
+   * @param original an original String.
+   * @return List of Doubles
+   */
+  public static List<Double> parseToDoubles(String original) {
+    List<Double> doubles = new ArrayList<>();
+
+    if (original != null) {
+      String[] array = original.split(SEMICOLON);
+      for (String radius : array) {
+        if (radius != null && !radius.isEmpty()) {
```

Comment:
We should add empty strings as `nulls`, otherwise, we are breaking the order of values in reference to the number of coordinates. Same in other occurrences. We should add tests around these scenarios.

Reply:
Yeah, as mentioned we should double check the formatting methods as there are cases in which `null` is valid and gets formatted to `""`.